### PR TITLE
Support custom registry on pushing image

### PR DESCRIPTION
**Pull request by lexcao**
Add feature by #1369

Need merge #1411 first for testing on absolute compose file.

### CODE REVIEW

**Comment by hangyan** [comment]
Will review this after #1411 is merged

**Comment by hangyan** [comment]
@lexcao Please rebase the master and push code again

**Comment by lexcao** [comment]
Because of inconvenient authentication for automatic test, I added a test file `script/test/cmd/tests_push_image.sh` which could test pushing image locally.

**Comment by hangyan** [comment]
related to #911

**Comment by lexcao** [comment]
@hangyan Thanks, I encountered the same problem on macOS, and resolve correctly by the given way.
Then I added a user guide on build and push image. Please help me on document sentences and expressions。

File: docs/user-guide.md [hunk 1]
```diff
@@ -390,6 +390,30 @@ If the Docker Compose file has service name with `_` or `.` in it (eg.`web_servi
 
 Please note that changing service name might break some `docker-compose` files.
 
+## Build and push image
+
+If the Docker Compose file has `build` or `build:context, build:dockerfile` keys, build will run when `--build` specified.
+
+And Image will push to *docker.io* (default) when `--push-image=true` specified.
+
+It is possible to push to custom registry by `--push-image-registry` argument, which will override the registry from image name.
```
Comment (hangyan) [inline docs/user-guide.md:399]: by `--push-image-registry` -> by specify `--push-image-registry`

File: docs/user-guide.md [hunk 2]
```diff
@@ -390,6 +390,30 @@ If the Docker Compose file has service name with `_` or `.` in it (eg.`web_servi
...
+
+And Image will push to *docker.io* (default) when `--push-image=true` specified.
+
+It is possible to push to custom registry by `--push-image-registry` argument, which will override the registry from image name. 
+
+### Authentication on registry
+
+Kompose use the docker authentication from file `$DOCKER_CONFIG/config.json`, `$HOME/.docker/config.json` and `$HOME/.dockercfg` after `docker login`.
+
+**This only works fine on Linux but macOS.**
```
Comment (hangyan) [inline docs/user-guide.md:405]: but macOS?

**Comment by lexcao** [comment]
@hangyan I have updated the document.
I used `grammarly` to check grammar, hope that works.

**Pull request merged**